### PR TITLE
Update samples generation

### PR DESCRIPTION
**Pull request by hadalin**
Changed argument names due to different terminology (samples/presamples) and did some cosmetic fixes.

Sample generation command now works, but I'm not sure if everything is ok, because I'm not familiar which fields have been added/updated. Maybe @mstajdohar should do a review pass.

### CODE REVIEW

File: resolwe_bio/management/commands/generate_samples.py [hunk 1]
```diff
@@ -21,15 +23,15 @@ class Command(BaseCommand):
     help = "Generate test data"
 
     def add_arguments(self, parser):
-        parser.add_argument('-a', '--n-annotated', type=int, default=15,
-                            help="Number of annotated samples to generate (default: %(default)s)")
-        parser.add_argument('-u', '--n-unannotated', type=int, default=5,
-                            help="Number of unannotated samples to generate (default: %(default)s)")
+        parser.add_argument('-s', '--n-samples', type=int, default=15,
+                            help="Number of samples to generate (default: %(default)s)")
+        parser.add_argument('-p', '--n-presamples', type=int, default=5,
```
Comment (hadalin) [inline resolwe_bio/management/commands/generate_samples.py]: Is this ok?

File: resolwe_bio/management/commands/generate_samples.py [hunk 2]
```diff
@@ -67,15 +70,22 @@ def generate_expressions(self, gene_ids, path):
...
+    def generate_reads_descriptor(annotated):
         barcodes = ['ATGCATGC', 'TACGTACG']
         instruments = ['Illumina HiSeq X', 'Illumina HiSeq 3000', 'Illumina HiSeq 3000']
         descriptor = {'barcode': random.choice(barcodes),
                       'instrument_type': random.choice(instruments),
                       'barcode_removed': True}
+
+        if annotated:
+            seq_date = datetime.date.today().strftime('%Y-%m-%d')
+            descriptor['seq_date'] = seq_date
```
Comment (hadalin) [inline resolwe_bio/management/commands/generate_samples.py]: Added `seq_date`. Only annotated samples should have this, right?

File: resolwe_bio/management/commands/generate_samples.py [hunk 3]
```diff
@@ -147,15 +157,18 @@ def create_data(self, reads_name='seq_reads', annotated=False, rseed=None):
...
-            descriptor_schema = DescriptorSchema.objects.get(slug='reads'),
-            descriptor = self.generate_reads_descriptor(),
-            status = 'OK',
-            process = Process.objects.get(slug='upload-fastq-single'),
-            contributor = self.set_user(),
-            input = {'src': {'file': os.path.basename(reads)}},
-            output = {
+            name=self.set_name(),
+            started=started,
+            finished=started + datetime.timedelta(minutes=45),
```
Comment (hadalin) [inline resolwe_bio/management/commands/generate_samples.py]: Populated `started` and `finished` fields.

File: resolwe_bio/management/commands/generate_samples.py (hunk 1 above)
Comment (tjanez) [inline resolwe_bio/management/commands/generate_samples.py]: I'm ok with this. @jkokosar, what do you think?

File: resolwe_bio/management/commands/generate_samples.py [hunk 4]
```diff
@@ -67,15 +70,22 @@ def generate_expressions(self, gene_ids, path):
...
-    def generate_reads_descriptor(self):
+    @staticmethod
+    def generate_reads_descriptor(annotated):
         barcodes = ['ATGCATGC', 'TACGTACG']
         instruments = ['Illumina HiSeq X', 'Illumina HiSeq 3000', 'Illumina HiSeq 3000']
         descriptor = {'barcode': random.choice(barcodes),
                       'instrument_type': random.choice(instruments),
                       'barcode_removed': True}
+
+        if annotated:
```
Comment (mstajdohar) [inline resolwe_bio/management/commands/generate_samples.py]: `seq_date` should be set for samples and presamples. In reality it is set by the script that uploads the files from the sequencer, this is the date of sequencing.

File: resolwe_bio/management/commands/generate_samples.py [hunk 5]
```diff
@@ -147,15 +157,18 @@ def create_data(self, reads_name='seq_reads', annotated=False, rseed=None):
         mouse_genes = os.path.join(test_files_path, 'mouse_genes.tab.gz')
 
         # crate reads data object
+        started = timezone.now()
```
Comment (mstajdohar) [inline resolwe_bio/management/commands/generate_samples.py]: Dates are always tricky. Please check if this work in the resolwe-bio test Django project that does not use time zones and in the genialis-base test Django project that uses time zones.

**Comment by mstajdohar** [comment]
@hadalin, 2 minor comment, otherwise good work.

**Comment by hadalin** [comment]
Tested. Works.

**Comment by hadalin** [comment]
Can this be merged?

**Comment by tjanez** [comment]
Since the current command doesn't work, I'm going to merge this. We'll have to continue working on improving this command in the future.

**Pull request merged**